### PR TITLE
Add as first argument to command the gitlab host to use to validate.

### PR DESCRIPTION
**Pull request by tpreviero**
Hi there 👋
I just needed to parametrize the gitlab host to use.
It might be useful, somehow 🙃

### CODE REVIEW

**Comment by Code0x58** [comment]
Thanks for your PR, it's nice to see others use this!

Would you mind amending this to use the [flag](https://gobyexample.com/command-line-flags) module? I'd be happy to merge something that was used like this:
```
gitlab-ci-validate file... # existing use
gitlab-ci-validate --host=http://my-gitlab.local file... # extended use
```
That way the existing way people use it isn't broken as the alternate server would is a nice additional option.

_P.S. on "useful, somehow :upside_down_face:": I think a use case for this would be if you can't/don't want to access the public service. The `.gitlab-ci.yml` format should be stable, but worst case there'd be an incompatible difference in validation between the public site and whatever instance you are running against (older or dev version)._

**Comment by tpreviero** [comment]
That's exactly the use-case I need to cover :sweat_smile: 
I'm pretty new to Go, but I'll do my best! I'll use this opportunity to learn a little.
Hoping to have some spare time during the weekend to work on it.

File: gitlab-ci-validate.go [hunk 1]
```diff
@@ -53,7 +54,7 @@ func ValidateFile(path string) (Validation, []error) {
 	}
 
 	values := url.Values{"content": {string(data)}}
-	request, err := http.NewRequest("POST", "https://gitlab.com/api/v4/ci/lint", strings.NewReader(values.Encode()))
+	request, err := http.NewRequest("POST", fmt.Sprintf("https://%s/api/v4/ci/lint", host), strings.NewReader(values.Encode()))
```
Comment (Code0x58) [inline gitlab-ci-validate.go]: To up the flexibility, the protocol part of the URL should also come from `--host`, that way people can also use it with HTTP only installations, which may happen in local development or private networks.

Comment (tpreviero) [reply gitlab-ci-validate.go]: You're absolutely right.
But doing that I wouldn't like anymore the name `host`.
What about [`origin`](https://tools.ietf.org/html/rfc6454#section-4)?

Comment (Code0x58) [reply gitlab-ci-validate.go]: origin may be a bit technical for the name of something on a command line tool (and it excludes user info). Naming is hard, but at least `--host` or `--server` pass for me

**Comment by Code0x58** [comment]
Looks good to me!

Are you happy this is tested and ready for merging?

**Comment by tpreviero** [comment]
Well...there are no test on it so far but..."it works on my machine"! :sweat_smile:
I might send others MR next week to this repository :wink:

**Comment by Code0x58** [comment]
_"works on my machine"_ works for me, I'll go through tests too just to double check.

Thanks! I'll do some touch ups tonight and see about automated tests more than `go fmt` and `go vet`

**Comment by Code0x58** [comment]
This is now out in [v0.2.0](https://github.com/Code0x58/gitlab-ci-validate/releases/tag/v0.2.0)

**Pull request merged**